objects.lisp

```diff
@@ -707,3 +850,128 @@
+  (print-unreadable-object (account-warning stream :type T)
+    (with-accessors ((id id)
+                     (action action)
+                     (text text)) account-warning
+      (format stream
+              "~a action ~a text ~a"
+              id
+              action
+              text))))
+
+(define-entity appeal
+  (text)
+  (state))
+
+(defmethod print-object ((account-warning account-warning) stream)
+  (print-unreadable-object (account-warning stream :type T)
+      (format stream
+              "text: ~a state: ~a"
+              (text account-warning)
+              (state account-warning))))
+
+(define-entity partial-account-with-avatar
+  (id)
+  (account-name :field "acct")
+  (url)
+  (avatar)
+  (avatar-static)
+  (locked)
+  (display-name)
+  (bot))
+
+(define-entity notification-group
+  (group-key :field "group_key")
+  (notifications-count :field "notifications_count")
+  (kind :field "type")
+  (most-recent-notification-id :field "most_recent_notification_id")
+  (page-min-id :field "page_min_id" :nullable t)
+  (page-max-id :field "page_max_id" :nullable t)
+  (latest-page-notification-at :field "latest_page_notification_at" :nullable t)
+  (sample-account-ids :field "sample_account_ids")
+  (status-id :field "status_id" :nullable t)
+  (report :translate-with #'decode-report :nullable t)
+  (relationship-severance-event :field "event"
+                                :translate-with #'decode-relationship-severance-event
+                                :nullable t)
+  (moderation-warning :field "moderation_warning" :nullable t))
+
+(define-entity grouped-notifications-results
+  (accounts :translate-with #'decode-account)
+  (partial-accounts :field "partial_accounts"
+                    :translate-with #'decode-partial-account-with-avatar
+                    :nullable t)
+  (statuses :translate-with #'decode-status)
+  (notification-groups :field "notification-groups"
+                       :translate-with #'decode-notification-group))
+
+(defun %decode-notification-policy-summary (summary-data)
+  (when (hash-table-p summary-data)
+    (list :pending-requests-count (gethash "pending_requests_count" summary-data)
+          :pending-notifications-count (gethash "pending_notifications_count" summary-data))))
+
+(define-entity notification-policy
+  (for-not-following    :field "for_not_following" :translate-with #'to-keyword)
+  (for-not-followers    :field "for_not_followers" :translate-with #'to-keyword)
+  (for-new-accounts     :field "for_new_accounts" :translate-with #'to-keyword)
+  (for-private-mentions :field "for_private_mentions" :translate-with #'to-keyword)
+  (for-limited-accounts :field "for_limited_accounts" :translate-with #'to-keyword)
+  (summary :translate-with #'%decode-notification-policy-summary))
+
+(defmethod print-object ((notification-policy notification-policy) stream)
+  (with-accessors ((for-not-following    for-not-following)
```

Comment:
I don't understand why you use with-accessors everywhere in these print-object methods when you then end up only using the accessor once. It doesn't save any space, it in fact wastes space.

Reply:
No valid general reason, it just fits better in my head this way, i prefer using a single symbol instead of of '(function object)', can i leave the code this way or should i change it?

Reply:
Please clean it up if it's not too much trouble!

Reply:
OK, no problem! :+1:

Reply:
Starting to change one method for commit, Is this style OK?

https://github.com/cage2/tooter/blob/update-api/objects.lisp#L85

Reply:
Yeah that's fine!

Reply:
perfect! :+1: